### DISCUSSION TITLE
Is it possible to have an unbound field to populate an underlying table?

### DISCUSSION THREAD

**Discussion by techthrasher** [Budibase/budibase#12703] [Help] [answered]
I would like to have a user input a time span in terms of days, hours, and minutes (separate form fields) but upon save convert that input into just hours for saving into the table.  This will reduce the number of fields in the table and also allow me to manipulate the reports based on what a user determines is a "day" (12 hours, 8 hours, 6 hours?).  I am new to Budibase and cannot find anything that equates to an unbound field on a form.  Is this possible?  I appreciate any help you might offer!

**Comment by ConorWebb96** [comment, marked as answer]
Hey @techthrasher,

Yes this is something can achieve fairly easily with the usage of Javascript. Below is how i did this.

1. Firstly I created the table with 2 columns `name` and `total time`
<img width="100%" alt="Screenshot 2024-01-04 at 12 34 37" src="https://github.com/Budibase/budibase/assets/126772285/e2f8b8e5-ed41-467c-8321-97f572cdf69f">

2. Then created a home design screen
3. Added a form block component
4. Ejected the form block component 
5. Deleted the total time field
6. Added 3 new fields `days`, `hours` and `minutes`.

<img width="100%" alt="Screenshot 2024-01-04 at 12 34 53" src="https://github.com/Budibase/budibase/assets/126772285/4aba0ae1-f9a0-48ca-8670-8301d8957266">
7. Clicked on the button component and modified its actions
<img width="1512" alt="Screenshot 2024-01-04 at 12 35 20" src="https://github.com/Budibase/budibase/assets/126772285/a4016191-f93a-4de6-9e09-8cbd20ff2bac">
<img width="100%" alt="Screenshot 2024-01-04 at 12 35 28" src="https://github.com/Budibase/budibase/assets/126772285/adba474c-1663-47d3-8fd4-c8711d38fc81">
8. Changed the saved row actions. (Kept the name the same as its not changing.)
<img width="100%" alt="Screenshot 2024-01-04 at 12 35 35" src="https://github.com/Budibase/budibase/assets/126772285/e5300e05-f47d-4513-892f-ecf27d83a62d">
9. Updated total time with the outcome of the below JS function.

```js
function calculateTotalHours(days, hours, minutes) {
  const daysInHours = days * 24;
  const minutesInHours = Math.floor(minutes / 60);
  const totalHours = daysInHours + hours + minutesInHours;

  return totalHours;
}
const totalHours = calculateTotalHours($("Form.Fields.days"), $("Form.Fields.hours"), $("Form.Fields.hours"));
return `${totalHours}`;
```

Demo in action.
![20240104_123648](https://github.com/Budibase/budibase/assets/126772285/7f402708-61c3-4847-afb2-2e96f6054d22)

**Reply by techthrasher** [reply]
THANK YOU!!!!!  So incredibly helpful....
